docs/pages/example/snapshot.html

```diff
@@ -0,0 +1,102 @@
+<style>
+  html, body {
+    margin: 0;
+  }
+
+  #map {
+    position: absolute;
+    width:50%;
+    height:100%;
+    background: rgba(0,255,0,0.5)
+  }
+
+  #snapshot {
+    position: absolute;
+    left:50%;
+    background:rgba(0,0,255,0.5)
+  }
+
+  h1 {
+    color:white;
+    text-align:center;
+    position: relative;
+    top:50%;
+    transform: translateY(-50%);
+    margin:0;
+  }
+ 
+</style>
+
+<div id='map' class='window'></div>
+
+<canvas id='snapshot' class='window'></div>
+
+<script>
+
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/dark-v9',
+    center: {
+        lat: 37.4036,
+        lng: -122.0325
+    },
+    zoom: 10,
+    preserveDrawingBuffer: true// not guaranteed to work if false
+});
+
+var snapshot = document.querySelector('#snapshot');
+var ctx = snapshot.getContext('2d');
+snapshot.height = innerHeight;
+snapshot.width = innerWidth / 2;
+var text = 'Click on the map';
+ctx.font = Math.round(snapshot.width / 10) + 'px Arial';
+ctx.fillStyle = 'white';
+ctx.textAlign = 'center';
+ctx.textBaseline = 'middle';
+ctx.fillText(text, snapshot.width / 2, snapshot.height / 2);
+
+map.on('click', function() {
+    var png = map.getCanvas().toDataURL();
+    var copy = new Image();
+    copy.src = png;
+    copy.onload = function() {
+        ctx.drawImage(copy, 0, 0);
+        logo();
+        if (snapshot.width > 400) {
```

Comment:
The attribution should always be shown regardless of the image width, similar to [how the static API works](https://www.mapbox.com/api-documentation/#retrieve-a-static-map-from-a-style), shortening to just (c) OSM should be okay on very small images where that is all that will fit.

Reply:
a text attribution will now always be displayed on the map, the short version (c) OSM kicks in if the canvas width is < 300px